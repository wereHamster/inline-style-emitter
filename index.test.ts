/* global describe, it, expect */
/* tslint:disable whitespace */

declare var describe: any, it: any, expect: any;

import { styleRules, cssStyleDeclarationsToText } from "./index";


function testSnapshots1(inputs, f) {
    describe("snapshots", () =>
        inputs.forEach(input =>
            it(JSON.stringify(input), () =>
                expect(f(input)).toMatchSnapshot())));
}


describe("styleRules", () => {
    it("should return an empty list if the style is empty", () => {
        expect(styleRules({}).length).toBe(0);
    });
    it("should return a single rule when the object has only local style declarations", () => {
        expect(styleRules({color:"red"}).length).toBe(1);
    });
    it("should parse nested pseudo components", () => {
        expect(styleRules({color:"red",":hover":{}}).length).toBe(1);
        expect(styleRules({color:"red",":hover":{color:"blue"}}).length).toBe(2);
    });
    it("should parse media query conditoinals", () => {
        expect(styleRules({"@media (max-width:1px)":{}}).length).toBe(0);
        expect(styleRules({"@media (max-width:1px)":{color:"red"}}).length).toBe(1);
    });
    it("should ignore @keyframes and @font-face", () => {
        expect(styleRules({"@keyframes":{test:{"0%":{width:"1px"}}}}).length).toBe(0);
        expect(styleRules({"@font-face":{test:{}}}).length).toBe(0);
    });

    testSnapshots1(
        [ {}
        , {color:"red"}
        , {color:"red",":hover":{}}
        , {color:"red",":hover":{color:"blue"}}
        , {"@media (max-width:1px)":{}}
        , {"@media (max-width:1px)":{color:"red"}}
        , {"@keyframes":{test:{"0%":{width:"1px"}}}}
        , {"@font-face":{test:{}}}
        ], styleRules);
});



describe("cssStyleDeclarationsToText", () => {
    it("should return the CSS string for simple rules", () => {
        expect(cssStyleDeclarationsToText({color:"red"})).toBe("color:red");
    });
    it("should support multiple property values in an array", () => {
        expect(cssStyleDeclarationsToText({color:["red","blue"]})).toBe("color:red;color:blue");
    });

    testSnapshots1(
        [ {color:"red"}
        , {color:["red","blue"]}
        ], cssStyleDeclarationsToText);
});
