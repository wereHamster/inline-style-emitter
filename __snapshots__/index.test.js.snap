exports[`cssStyleDeclarationsToText snapshots {"color":"red"} 1`] = `"color:red"`;

exports[`cssStyleDeclarationsToText snapshots {"color":["red","blue"]} 1`] = `"color:red;color:blue"`;

exports[`styleRules snapshots {"@font-face":{"test":{}}} 1`] = `Array []`;

exports[`styleRules snapshots {"@keyframes":{"test":{"0%":{"width":"1px"}}}} 1`] = `Array []`;

exports[`styleRules snapshots {"@media (max-width:1px)":{"color":"red"}} 1`] = `
Array [
  Object {
    "conditions": Array [
      "@media (max-width:1px)"
    ],
    "style": Object {
      "color": "red"
    },
    "suffixes": Array []
  }
]
`;

exports[`styleRules snapshots {"@media (max-width:1px)":{}} 1`] = `Array []`;

exports[`styleRules snapshots {"color":"red",":hover":{"color":"blue"}} 1`] = `
Array [
  Object {
    "conditions": Array [],
    "style": Object {
      "color": "blue"
    },
    "suffixes": Array [
      ":hover"
    ]
  },
  Object {
    "conditions": Array [],
    "style": Object {
      "color": "red"
    },
    "suffixes": Array []
  }
]
`;

exports[`styleRules snapshots {"color":"red",":hover":{}} 1`] = `
Array [
  Object {
    "conditions": Array [],
    "style": Object {
      "color": "red"
    },
    "suffixes": Array []
  }
]
`;

exports[`styleRules snapshots {"color":"red"} 1`] = `
Array [
  Object {
    "conditions": Array [],
    "style": Object {
      "color": "red"
    },
    "suffixes": Array []
  }
]
`;

exports[`styleRules snapshots {} 1`] = `Array []`;
